.github/dependabot.yml

```diff
@@ -15,25 +15,29 @@ updates:
     ignore:
       - dependency-name: react
         versions:
-          - '^16.11.0'
+          - '>=18.0.0'
```

Comment:
Why did dependabot not ignore react 17 with this rule? Without reading the docs, it seems like it should have.

Reply:
From my reading [here](https://docs.github.com/en/github/administering-a-repository/configuration-options-for-dependency-updates#ignore), the `versions` option uses npm's semver to parse version ranges that should be ignored. `^16.11.0` is equivalent to a range of versions `>=16.11.0` and `<17.0.0`.

What we're unsure about is why this only just now happened. React 17 was released in October of last year. The timing is suspect because we also recently moved `react` to `devDependency`. From my limited research, it sounds like dependabot should treat `devDependencies` with the same config-derived logic, but I also didn't find an answer that felt definitive on that.

Reply:
From what I can tell from [the docs](https://docs.github.com/en/github/administering-a-repository/configuration-options-for-dependency-updates#allow) it seems we were only allowing development dependencies? That doesn't seem to make sense with the PRs we see, though. It does seem like we want `dependency-type: direct` though?

Reply:
@ahobson you're right, I think that is what caused dependabot to upgrade React to 17. tbh this project should only have development & peer dependencies to begin with, but changing `dependency-type: direct` shouldn't hurt.

Reply:
@suzubara I'll update that when I fix conflicts here  👍

Reply:
43222c2